deebotozmo/cli.py

```diff
@@ -0,0 +1,343 @@
+        print(json.dumps(asdict(event)))
+        lock.release()
+
+    listener = vacbot.bot.cleanLogsEvents.subscribe(on_clean_event)
+    await vacbot.bot.execute_command(GetCleanLogs())
+    await lock.acquire()
+    vacbot.bot.cleanLogsEvents.unsubscribe(listener)
+    await vacbot.goodbye()
+
+@cli.command(help='Get robot statuses [Status,Battery,FanSpeed,WaterLevel]')
+@coro
+async def statuses():
+    vacbot = dologin()
+    await vacbot.run()
+    lock = asyncio.Lock()
+
+    print ("Vacuum State: " + str(vacbot.bot.status.state).split(".")[-1])
+
+    await lock.acquire()
+    async def on_battery(event: BatteryEvent):
+        print("Battery: " + str(event.value) + "%")
+        lock.release()
+    listener = vacbot.bot.batteryEvents.subscribe(on_battery)
+    await vacbot.bot.execute_command(GetBattery())
+    await lock.acquire()
+    vacbot.bot.batteryEvents.unsubscribe(listener)
+
+    async def on_fan_event(event: FanSpeedEvent):
+        print("Fan Speed: " + str(event.speed))
+        lock.release()
+    listener = vacbot.bot.fanSpeedEvents.subscribe(on_fan_event)
+    await vacbot.bot.execute_command(GetFanSpeed())
+    await lock.acquire()
+    vacbot.bot.fanSpeedEvents.unsubscribe(listener)
+
+    async def on_water_level(event: WaterInfoEvent):
+        print("Water Level: " + str(event.amount))
+        lock.release()
+    listener = vacbot.bot.waterEvents.subscribe(on_water_level)
+    await vacbot.bot.execute_command(GetWaterInfo())
+    await lock.acquire()
+    vacbot.bot.waterEvents.unsubscribe(listener)
+
+    await vacbot.goodbye()
+
+@cli.command(help='Get stats')
+@coro
+async def stats():
+    vacbot = dologin()
+    await vacbot.run()
+
+    lock = asyncio.Lock()
+    await lock.acquire()
+    async def on_stats_event(event: StatsEvent):
+        print("Stats Cid: " + str(event.clean_id))
+        print("Stats Area: " + str(event.area))
+        print("Stats Time: " + str(int(event.time / 60)) + " minutes")
+        print("Stats Type: " + str(event.type))
+        lock.release()
+    listener = vacbot.bot.statsEvents.subscribe(on_stats_event)
+    await vacbot.bot.execute_command(GetStats())
+    await lock.acquire()
+    vacbot.bot.statsEvents.unsubscribe(listener)
+    await vacbot.goodbye()
+
+# Needs a lock but the lifespan event is emitted
+# for every component, so it wouldn't work
+@cli.command(help='Get robot components life span')
+@coro
+async def components():
+    vacbot = dologin()
+    await vacbot.run()
+
+    async def on_lifespan_event(event: LifeSpanEvent):
+        print(str(event.type) + ": " + str(event.percent) + "%")
+    listener = vacbot.bot.lifespanEvents.subscribe(on_lifespan_event)
+    await vacbot.bot.execute_command(GetLifeSpan())
+    vacbot.bot.lifespanEvents.unsubscribe(listener)
+    await vacbot.goodbye()
+
+@cli.command(help='Get saved rooms')
+@coro
+async def getrooms():
+    print ("Not supported, no clue how to make it work.")
```

Comment:
Use `listener = vacbot.bot.map.roomsEvents.subscribe(on_rooms)`
Similar to the other events